src/Engine/Renderer/RenderObject/RenderObject.cpp

```diff
@@ -106,7 +106,8 @@ namespace Ra
             for (size_t i = 0; i < asset->getVerticesSize(); ++i)
             {
                 mesh.m_vertices.push_back(T * asset->getVertices()[i]);
-                mesh.m_normals.push_back((N * asset->getNormals()[i]).normalized());
+                if(asset->hasNormals())
+                    mesh.m_normals.push_back((N * asset->getNormals()[i]).normalized());
```

Comment:
Could you please reserve std::vectors capacity before calling push_back in for loops ?